lib/game_runner.rb

```diff
@@ -49,11 +55,16 @@ def print_grid
     grid_printer.print_grid
   end
 
+  def mark_grid_with_player_input
+    grid.mark(get_valid_player_coordinate, 'X')
```

Comment:
I'm thinking that the grid should know about the X and 0. It can have methods for the computer and the player, and that data about the right mark can be encapsulated there instead of here.

If you really think about it, there are two players and each has a mark. I wonder what it would look like to have two player, each who knew how to get a valid coordinate and mark the grid. That would be another approach to encapsulating the logic.

Reply:
Got it. I think I've addressed this in my 2nd take